exercises/047_methods.zig

```diff
@@ -63,7 +66,9 @@ const HeatRay = struct {
 
     // We love this method:
     pub fn zap(self: HeatRay, alien: *Alien) void {
-        alien.health -= if (self.damage >= alien.health) alien.health else self.damage;
+        alien.health -|= self.damage; // Saturating inplace substraction
+                                      // It subtracts but doesn't go below the
+                                      // lowest value for our type (in this case 0)
```

Comment:
Wow, I didn't even _know_ about the saturating mathematical operators. When did those get added?  Ha ha. That's really cool.

I'm afraid we would need to cover them before we use that in this exercise. I don't want this to be the first time the learner sees it. It's a tough balance. I've had problems trying to jam too many concepts into earlier exercises.

Reply:
I think the Zig stance is that at overflow or underflow you panic by default and have explicit operators for wrapping and saturating.

Yea, the more I think about it this change was just me showing off the language and not teaching it.